File: Ridy-taxi-app/ios/common/Models/Objects/Media.swift
//
//  Media.swift
//
//  Copyright (c) Minimalistic apps. All rights reserved.
//

import Foundation

public final class Media: Codable {

    public enum PathType: String, Codable {
        case absolute = "absolute"
        case relative = "relative"
    }
    public enum PrivacyLevel: String, Codable {
        case low = "low"
        case medium = "medium"
        case high = "high"
    }
  // MARK: Declaration for string constants to be used to decode and also serialize.
  enum CodingKeys: String, CodingKey {
    case id = "id"
    case address = "address"
    case privacyLevel = "privacy_level"
    case title = "title"
    case type = "type"
    case pathType = "path_type"
  }

  // MARK: Properties
  public var id: Int?
  public var address: String?
  public var privacyLevel: PrivacyLevel?
  public var title: String?
  public var type: String?
  public var pathType: PathType?
}

File: Ridy-taxi-app/android/common/src/main/java/com/innomalist/taxi/common/models/Media.kt
package com.innomalist.taxi.common.models

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass
import java.io.Serializable
@JsonClass(generateAdapter = true)
class Media(var address: String, var pathType: PathType) : Serializable {
    var id = 0

    enum class PathType {
        @Json(name="relative")
        relative,
        @Json(name="absolute")
        absolute
    }

    var privacyLevel: String? = null
    var title: String? = null

}